### PR TITLE
Drivers: SPI: Add support for STM32L4

### PR DESCRIPTION
**Pull request by puranjaymohan**
Add configuration files for STM32L4 related to SPI
Make changes in HAL configurations to add SPI support
Change some other files for different reasons.

Signed-off-by: Puranjay Mohan <puranjay12@gmail.com>

### CODE REVIEW

File: src/drivers/char/char_dev_stub.h [hunk 1]
```diff
@@ -20,7 +20,7 @@
 		__attribute__((unused)) static struct { \
 			void *o, *c, *i, *p; \
 		} unused##__LINE__ = { \
-			.o = open_fn, .c = close_fn, .i = idesc_op, .p = priv \
```
Comment (anton-bondarev) [inline src/drivers/char/char_dev_stub.h:23]: Could you please extract the change to a separate commit?

Comment (puranjaymohan) [reply src/drivers/char/char_dev_stub.h]: I have added a separate PR for this and removed it from here.

**Comment by anton-bondarev** [comment]
@puranjaymohan Thanks a lot
I've merged it.
